### PR TITLE
Use checkmate package to assert args

### PR DESCRIPTION
**Pull request by artemklevtsov**
Replace all `stopifnot` with `checkmate::assert_*`.
Replace `is_string_len_one` with `checkmate::assert_string`.
Replace `is_string_or_null` with `checkmate::assert_string(..., null.ok = TRUE)`.
Refactor `RestRserveApplication$add_static` for the clear and simplify. Added `private$guess_mime` method.
Refactor `RestRserveApplication$call_handler` to optimize matching path.
Replace all `require` calls with `requireNamespace` as canonical way to check a package installed.

### CODE REVIEW

**Comment by eddelbuettel** [comment]
(Minor typo in Subject:  checkmate, with an e in check.  Nice package.)

File: R/RestRserveApplication.R [hunk 1]
```diff
@@ -226,36 +197,44 @@ RestRserveApplication = R6::R6Class(
       FUN = private$handlers[[request$path]][[request$method]]
 
       if(!is.null(FUN)) {
-        self$logger$trace(list(request_id = request$request_id, message = 'exact endpoint match for the route'))
+        msg <- "exact endpoint match for the route"
+        private$logger$trace(list(request_id = request$request_id, message = msg))
       } else {
-        self$logger$trace(list(request_id = request$request_id, message = "haven't found exact endpoint match for requested route"))
+        msg <- "haven't found exact endpoint match for requested route"
```
Comment (dselivanov) [inline R/RestRserveApplication.R]: we prefer `=` over `<-`

File: R/RestRserveApplication.R [hunk 2]
```diff
@@ -74,8 +74,11 @@ RestRserveApplication = R6::R6Class(
     HTTPError = NULL,
     #------------------------------------------------------------------------
     initialize = function(middleware = list(),
-                          content_type = "application/json", ...) {
+                          content_type = "text/plain",
+                          serializer = NULL,
+                          ...) {
       checkmate::assert_list(middleware)
+      self$HTTPError = HTTPErrorFactory$new(content_type, serializer)
       dots = list(...)
```
Comment (artemklevtsov) [inline R/RestRserveApplication.R]: В данном случае я бы использовал `hasArg("logger")`.

File: R/RestRserveResponse.R [hunk 3]
```diff
@@ -63,6 +63,8 @@ RestRserveResponse = R6::R6Class(
                           content_type = 'text/plain',
                           headers = character(0),
                           status_code = 200L) {
+                          serializer = NULL) {
+      self$set_content_type(content_type, serializer)
```
Comment (artemklevtsov) [inline R/RestRserveResponse.R]: Возможно стоит опустить после проверок.

File: R/RestRserveApplication.R [hunk 4]
```diff
@@ -79,13 +79,12 @@ RestRserveApplication = R6::R6Class(
...
       self$HTTPError = HTTPErrorFactory$new(content_type, serializer)
-      dots = list(...)
-      if("logger" %in% names(dots)) {
+      if(hasArg("logger")) {
         msg = paste("THIS MESSAGE WILL BE TURNED INTO ERROR SOON",
                     "'logger' argument is DEPRECATED, please use public `app$logger` field to control logging.",
                     sep = "\n")
         warning(msg)
-        self$logger = dots$logger
+        self$logger = logger
```
Comment (dselivanov) [inline R/RestRserveApplication.R]: `list(...)$logger` ?

Comment (artemklevtsov) [reply R/RestRserveApplication.R]: Seems you're right. Also we need check for the `methods` namespace import.

**Pull request merged**